etc/compile_check.sh

```diff
@@ -0,0 +1,79 @@
+#!/usr/bin/env bash
+set -e # exit when any command fails
+
+GC=go
+WD=$(pwd)
+COMPILE_CHECK_DIR="internal/test/compilecheck"
+DEV_MIN_VERSION=1.19
+
+# version will flatten a version string of upto 4 components for inequality
+# comparison.
+function version {
+	echo "$@" | awk -F. '{ printf("%d%03d%03d%03d\n", $1,$2,$3,$4); }';
+}
+
+# dev_compile_check will attempt to build in the development environment. This
+# check will only run on environments where the Go version is greater than or
+# equal to the DEV_MIN_VERSION.
+function dev_compile_check {
+	MACHINE_VERSION=`${GC} version | { read _ _ v _; echo ${v#go}; }`
+
+	if [ $(version $MACHINE_VERSION) -lt $(version $DEV_MIN_VERSION) ]; then
+		return
+	fi
```

Comment:
I'm concerned this logic could lead to builds that pass without actually running anything. For example, if a dev is using Go 1.18, they may run this compile check script and assume that everything worked fine because it didn't return any errors. Instead, they should see any relevant compilation errors.

Reply:
Now that `compile_check` runs most commands with any version of Go, I think this is safer to keep. However, as noted in my other comment, `dev_compile_check` seems redundant with other Makefile targets now and could probably be removed. Consider this comment thread resolved, tho.